### DISCUSSION TITLE
Cannot parse the partition key from SQL for less table scan if the partition keys are specified in OR conditions

### DISCUSSION THREAD

**Discussion by pangyifish** [trinodb/trino#13467] [General]
`a` is a hive table partitioned by `dt`, stored in s3.
I have where conditions like:

```
WHERE
    cg.dt = DATE'2022-07-24'
                                AND (
                                    (a.dt = DATE'2022-07-17' AND a."time" >= cg.time - 604800000000)
                                    OR 
                                    (a.dt = DATE'2022-07-18'AND a."time" < 1658188800000000 - 300000000)
                                    OR
                                    (a.dt = DATE'2022-07-24' AND a."time" < cg.time - 300000000)
                                    OR  (a.dt = DATE'2022-07-23' AND a."time" >= 1658620800000000 - 300000000 AND a."time" < cg.time - 300000000)
                               )
                                AND a.id IS NOT NULL
                                AND a.id > 0
```

Trino ended up doing a full table scan on `a`
so I have to add `AND a.dt in (DATE'2022-07-17', DATE'2022-07-18', DATE'2022-07-23', DATE'2022-07-24')` to make it scan less data. 

version: 380

**Comment by findepi** [comment]
This is a known limitation of TupleDomain-based predicate pushdown.
This is why we recently implemented expression-based predicate pushdown, see PostgreSQL connector for example.

Same technique could be applied in other connectors as well (eg Iceberg), but seems not quite applicable to eg Hive.

**Reply by pangyifish** [reply]
Thanks! Do we have plans to address this for Hive connector in the future?
